src/js/tetris.js

```diff
@@ -0,0 +1,235 @@
+import I from './i';
+import J from './j';
+import L from './l';
+import O from './o';
+import S from './s';
+import T from './t';
+import Z from './z';
+import Block from './block';
+import { CONSTANTS } from './values';
+
+const SHAPES = [I, J, L, O, S, T, Z];
+const KEYS = {
+  left: 37,
+  right: 39,
+  down: 40,
+  r: 82,
+  space: 32
+};
+function sleep(time) {
+  return new Promise(resolve => setTimeout(resolve, time));
+}
+
+export default class Tetris {
+  constructor({ container, unitSize }) {
+    this.container = container;
+    this.unitSize = unitSize;
+    this.blocks = [];
+    this.round = 1;
+    this.pause = false;
+    this.speed = 1000;
+    this.score = 0;
+    this.isRunning = false;
+    this.highScore = 0;
+  }
+
+  get elements() {
+    return {
+      score: document.querySelector('.score__value'),
+      highScore: document.querySelector('.high-score__value'),
+      nextShape: document.querySelector('.next-shape__object'),
+      pause: document.querySelector('.pause'),
+      tetris: document.querySelector('.tetris'),
+      gameOver: document.querySelector('.game-over')
+    };
+  }
+
+  start() {
+    this.isRunning = true;
+    if (!this.gameOver) {
+      this.setupListeners();
+    }
+    this.gameOver = false;
+    this.shape = this.getRandomShape();
+    this.shape.draw(this.container);
+    this.drawNextShape();
+    this.loop();
```

Comment:
Instead of having this loop function handle all the drawing and checking, we can isolate the logic more. 
We can do something like a `drawShape` function here whose job is to 
1- Determine were it is going to be rendered
2- Draw itself into the grid
3- Move itself.

So the shape instance should have a `moveDown` method that 
1- Checks if it can move down
2- Moves down if possible

```
drawShape() {
  1- set y position on the shape
  2 - draw the shape
  3 - moveCurrentShape();
}

moveCurrentShape() {
  1- move shape
  2- if can move shape calls itself again => here we can do a setTimeout which can be determined by level/speed.
  3- If cannot move shape calls gameOver or draws next shape
}
```